templates/docs/examples/patterns/navigation/search-dark.html

```diff
@@ -0,0 +1,109 @@
+{% extends "_layouts/examples.html" %}
+
+{% block title %}Navigation / Search dark{% endblock %}
+
+{% block standalone_css %}patterns_navigation{% endblock %}
+
+{% block content %}
+
+<header id="navigation" class="p-navigation is-dark">
+  <div class="p-navigation__row">
+    <div class="p-navigation__banner">
+      <div class="p-navigation__logo">
+        <a class="p-navigation__item" href="#">
+          <img class="p-navigation__image" src="https://assets.ubuntu.com/v1/3c7954dd-logo-canonical-white.svg" alt="" width="95" />
+        </a>
+      </div>
+      <a href="#navigation" class="p-navigation__toggle--open" title="menu">Menu</a>
+      <a href="#navigation-closed" class="p-navigation__toggle--close" title="close menu">Close menu</a>
+    </div>
+    <nav class="p-navigation__nav" aria-label="Example main navigation">
+      <ul class="p-navigation__items">
+        <li class="p-navigation__item is-selected">
+          <a class="p-navigation__link" href="#">Products</a>
+        </li>
+        <li class="p-navigation__item">
+          <a class="p-navigation__link" href="#">Services</a>
+        </li>
+        <li class="p-navigation__item">
+          <a class="p-navigation__link" href="#">Partners</a>
+        </li>
+        <li class="p-navigation__item">
+          <a class="p-navigation__link" href="#">About</a>
+        </li>
+			</ul>
+			<ul class="p-navigation__items">
+				<li class="p-navigation__item" id="link-4">
+					<a href="#" class="js-search-button p-navigation__link">
+						<span class="u-hide u-show--large">Search</span> <i class="p-icon--search is-light"></i>
+					</a>
+        </li>
+			</ul> 
+      <div class="p-navigation__search u-show--small u-hide" style="z-index: 39;">
+        <form action="/search" class="p-search-box" id="ubuntu-global-search-form">
+          <input type="search" class="p-search-box__input" name="q" placeholder="Search our sites" required="" aria-label="Search our sites">
+          <button type="reset" class="p-search-box__reset"><i class="p-icon--close"></i></button>
+          <button type="submit" class="p-search-box__button"><i class="p-icon--search">Search</i></button>
+        </form>
+      </div>
+    </nav>
+  </div>
+</header>
+
+<script>
+	function initSearch() {
+		console.log("clicked")
+    var searchButton = document.querySelector('.js-search-button');
+    var searchReset = document.querySelector('.p-search-box__reset');
+    if (searchButton) {
+      searchButton.addEventListener('click', openSearch);
+    }
+
+    if (searchReset) {
+      searchReset.addEventListener('click', closeSearch);
+    }
+  }
+  initSearch();
+
+  function openSearch(e) {
+		console.log("clicked")
+    e.preventDefault();
+    var navigation = document.querySelector('.p-navigation__nav');
```

Comment:
There is a lot of class names hardcoded in this script, adding class names and changing styles of various elements. I think it would be worth refactoring it a bit to have one class name (maybe `has-search-open`) on root element that will affect child elements (hide, show, etc).

We can discuss it separately.

Reply:
Hey Bartek I tried to do that, have a look please